### PR TITLE
destroy operator with force option

### PR DESCRIPTION
**Pull request by msenmurugan**

### CODE REVIEW

**Comment by operator-r2d2** [comment]
[APPROVALNOTIFIER] This PR is **APPROVED**

This pull-request has been approved by: *<a href="https://github.com/blackducksoftware/synopsys-operator/pull/527#" title="Author self-approved">msenmurugan</a>*

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=blackducksoftware%2Fsynopsys-operator).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details >
Needs approval from an approver in each of these files:

- ~~[OWNERS](https://github.com/blackducksoftware/synopsys-operator/blob/master/OWNERS)~~ [msenmurugan]

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Comment by operator-r2d2** [comment]
@msenmurugan: once the present PR merges, I will cherry-pick it on top of release-2019.6.x in a new PR and assign it to you.

<details>

In response to [this](https://github.com/blackducksoftware/synopsys-operator/pull/527#issuecomment-504727575):

>/cherrypick release-2019.6.x


Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository.
</details>

File: pkg/synopsysctl/cmd_destroy.go [hunk 1]
```diff
@@ -88,9 +92,26 @@ func destroy(namespace string) {
...
 
 	// delete namespace
-	err = util.DeleteResourceNamespace(restconfig, kubeClient, strings.Join(crds, ","), namespace, true)
-	if err != nil {
-		log.Warn(err)
+	checkErr := util.CheckResourceNamespace(restconfig, kubeClient, strings.Join(crds, ","), namespace, true)
+	if err != nil && isForceDestroy {
+		log.Warnf("%s. namespace cannot be deleted", err.Error())
+	} else if err != nil && !isForceDestroy {
+		return fmt.Errorf("%s. we do not recommend destroying the Synopsys Operator for proper upkeep of these resources. you can use 'force' option, which will keep the instances, but delete the Synopsys Operator.  You can re-deploy the Synopsys Operator in this namespace at any time", err.Error())
```
Comment (rrati) [inline pkg/synopsysctl/cmd_destroy.go]: we do not ...-> it is not recommended to destroy the Synopsys Operator so these resources can continue to be managed.  If you are sure you want to delete the Synopsys Operator anyway then you can use the 'force' option which will keep all the instances and delete only the Synopsys Operator. ...

**Comment by operator-r2d2** [comment]
[APPROVALNOTIFIER] This PR is **APPROVED**

This pull-request has been approved by: *<a href="https://github.com/blackducksoftware/synopsys-operator/pull/527#" title="Author self-approved">msenmurugan</a>*, *<a href="https://github.com/blackducksoftware/synopsys-operator/pull/527#pullrequestreview-253391925" title="Approved">rrati</a>*

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=blackducksoftware%2Fsynopsys-operator).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details >
Needs approval from an approver in each of these files:

- ~~[OWNERS](https://github.com/blackducksoftware/synopsys-operator/blob/master/OWNERS)~~ [msenmurugan,rrati]

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

File: pkg/synopsysctl/cmd_destroy.go (hunk 1 above)
Comment (msenmurugan) [reply pkg/synopsysctl/cmd_destroy.go]: @rrati done Rob

**Comment by operator-r2d2** [comment]
[APPROVALNOTIFIER] This PR is **APPROVED**

This pull-request has been approved by: *<a href="https://github.com/blackducksoftware/synopsys-operator/pull/527#" title="Author self-approved">msenmurugan</a>*, *<a href="https://github.com/blackducksoftware/synopsys-operator/pull/527#pullrequestreview-253452011" title="LGTM">rrati</a>*

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=blackducksoftware%2Fsynopsys-operator).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details >
Needs approval from an approver in each of these files:

- ~~[OWNERS](https://github.com/blackducksoftware/synopsys-operator/blob/master/OWNERS)~~ [msenmurugan,rrati]

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Comment by operator-r2d2** [comment]
@msenmurugan: new pull request created: #531

<details>

In response to [this](https://github.com/blackducksoftware/synopsys-operator/pull/527#issuecomment-504727575):

>/cherrypick release-2019.6.x


Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository.
</details>

**Pull request merged**